### PR TITLE
Fix issues with frameworks like Laravel on PHP7

### PR DESCRIPTION
**Pull request by xwiz**
Allow exception or throwable to be passed to whoops

### CODE REVIEW

File: src/Whoops/Exception/Inspector.php [hunk 1]
```diff
@@ -58,7 +58,7 @@ public function getExceptionMessage()
     }
 
     /**
-     * Does the wrapped Exception has a previous Exception?
+     * Does the wrapped mixed have a previous Exception?
```
Comment (hultberg) [inline src/Whoops/Exception/Inspector.php]: Hum, is this an intended change?

File: src/Whoops/Exception/Inspector.php [hunk 2]
```diff
@@ -6,12 +6,12 @@
 
 namespace Whoops\Exception;
 
-use Exception;
+use Throwable;
```
Comment (prisis) [inline src/Whoops/Exception/Inspector.php]: Wrong | no need for this

File: src/Whoops/Exception/Inspector.php [hunk 3]
```diff
@@ -6,12 +6,12 @@
...
 
 class Inspector
 {
     /**
-     * @var Exception
+     * @var Throwable
```
Comment (prisis) [inline src/Whoops/Exception/Inspector.php]: Use \Exception | \Throwable

File: src/Whoops/Exception/Inspector.php [hunk 4]
```diff
@@ -26,15 +26,15 @@ class Inspector
     private $previousExceptionInspector;
 
     /**
-     * @param Exception $exception The exception to inspect
+     * @param mixed $exception The exception to inspect
```
Comment (prisis) [inline src/Whoops/Exception/Inspector.php]: Wrong use \Exception | \Throwable

**Comment by prisis** [comment]
Hey @xwiz you pr is wrong you need to remove all mixed and change it to \Exception | \Throwable and only remove Exception in functions.

File: src/Whoops/Exception/Inspector.php (hunk 1 above)
Comment (xwiz) [inline src/Whoops/Exception/Inspector.php]: Ah, thanks for the catch.. Pushed changes now...

**Comment by prisis** [comment]
But wait a min, why you dont use the v2?

**Comment by GrahamCampbell** [comment]
This is a major breaking change.

**Comment by GrahamCampbell** [comment]
That's why it only happened on v2.

**Comment by GrahamCampbell** [comment]
Upgrade to newer laravel to use php 7. Laravel 5.1 or upwards is needed.

**Comment by GrahamCampbell** [comment]
> Fix issues with frameworks like Laravel on PHP7

Laravel 4 does NOT support PHP 7.

**Comment by xwiz** [comment]
@prisis v1 is included by Laravel 4.2 by default. Technically, I don't see anything wrong with Laravel 4.2 and many have gone as far as optimizing it by including their own packages and stuff. I don't see why anyone who decides or wants to continue using it can't use it.

**Comment by xwiz** [comment]
@GrahamCampbell how do you mean it's a major breaking change? What kind of test would fail?

**Comment by GrahamCampbell** [comment]
Anyone extending this class will break.

**Comment by GrahamCampbell** [comment]
> Technically, I don't see anything wrong with Laravel 4.2 

I can guarantee you L4.2 does not work properly on PHP 7. I made the needed changes to Laravel my self to get L5.1 to work.

**Comment by xwiz** [comment]
@GrahamCampbell Interesting. I've heard that it's not compatible, but I don't exactly know what's not compatible except for the error handling which is pretty easy to resolve if you edit the files directly.

**Comment by GrahamCampbell** [comment]
There are about 40 places that need to change to get it to work.

**Comment by xwiz** [comment]
Wow hmmm, thanks for the information. Do you have a fork I can look at? Would be very helpful

**Comment by GrahamCampbell** [comment]
Sorry, no. I'd recommend upgrading to L5.1 LTS.

**Comment by xwiz** [comment]
@GrahamCampbell Me as well. I just wanted to see out of curiosity. Also I prefer 5.2

**Comment by xwiz** [comment]
One issue I don't like with Laravel 5 is the injected/default Validator. I think that belongs to the base Model class, and Laravel should also support defining rules via arrays at the base model class. This is me just saying, not like I've taken it up. I believe every developer should do whatever they want with a framework but there're ways that can be argued to be better in achieving an aim.

**Comment by prisis** [comment]
@xwiz can you close this pr?

**Comment by GrahamCampbell** [comment]
> Also I prefer 5.2

I meant upgrade to at least 5.1.

**Pull request closed without merging**